tests/test_fakes.py

```diff
@@ -290,6 +290,55 @@ def test_connect_with_non_existent_db_or_schema(_fakesnow_no_auto_create: None):
         assert conn.schema == "JAFFLES"
 
 
+def test_dateadd_date_cast(conn: snowflake.connector.SnowflakeConnection):
+    with conn.cursor(snowflake.connector.DictCursor) as cur:
+        cur.execute("create table example (d varchar)")
+        cur.execute("insert into example values ('2023-04-02')")
+
+        q = """
+        SELECT
+            dateadd(hour, 1, d::date) as d_noop,
+            dateadd(day, 1, d::date) as d_day,
+            dateadd(week, 1, d::date) as d_week,
+            dateadd(month, 1, d::date) as d_month,
+            dateadd(year, 1, d::date) as d_year
+        FROM example
+        """
+
+        cur.execute(q)
+        assert cur.fetchall() == [
+            {
+                "D_NOOP": datetime.datetime(2023, 4, 2, 1, 0),
+                "D_DAY": datetime.date(2023, 4, 3),
+                "D_WEEK": datetime.date(2023, 4, 9),
+                "D_MONTH": datetime.date(2023, 5, 2),
+                "D_YEAR": datetime.date(2024, 4, 2),
+            }
+        ]
```

Comment:
probably not strictly needed, as the test below will cover this already
```suggestion
```

Reply:
sorry i was on offline for quite some time.